docs/enhancements/add-kernel-parameters.md

```diff
@@ -0,0 +1,101 @@
+---
+title: add-discovery-kernel-arguments
+authors:
+- "@oamizur"
+  creation-date: 2022-09-20
+  last-updated: 2022-10-23
+---
+
+# Add discovery kernel arguments
+
+## Summary
+
+Some environments require additional kernel arguments to boot host machines.  
+We would like to support additional kernel arguments while booting the discovery image.
+
+## Motivation
+
+Previously, adding kernel arguments was required by several customers in order to adapt to various environments.
+For example **rd.net.timeout.carrier** may be required in order for either static networking to be applied or 
+to successfully get a dhcp address before pulling the rootfs.
+### Goals
+Add capability to set additional kernel arguments to discovery image for append.
+
+- Image types: Full ISO, minimal ISO, iPXE
+- Deployment types: Cloud, operator
+
+API which supports full set of operations as defined by Coreos Installer: append, replace, delete.
+
+### Non-Goals
+Manipulation of existing kernel parameters (replace/delete).  Replace and delete of exiting
+kernel arguments in ISO file is supported by Coreos Installer, but this implementation will not 
+support it. 
+
+## Proposal
+
+The discovery kernel arguments will be stored in the infra-env database record in a designated JSON encoded field.
+During infra-env registration and update the full set of the kernel arguments can be set.
+
+Setting the arguments can either be done by either REST API or Kube API (Part of InfraEnv CR)
+### User Stories
+
+#### Story 1
+
+When having slow NIC, increase the timeout to have the IP address configured.
+
+### Implementation Details/Notes/Constraints
+
+#### iPXE implementation
+
+iPXE implementation should be quite straight forward.  The arguments will be concatenated at the end of the 
+**kernel** command which is part of the iPXE script.
+
+#### Minimal and full ISO
+
+When streaming the discovery ISO file, the kernel arguments are included
+as part of some files: grub.cfg and isolinux.cfg
+
+Each of these files contains an area where the kernel arguments can be
+embedded. This can be identified by the regex **"(\n#+) COREOS_KARG_EMBED_AREA"**.
+The arguments should overwrite this are from the beginning and should
+not pass the last # (hash sign). The added kernel will be injected in the beginning.
+These arguments will be seperated by space **(" ")**.  When embedding the kernel arguments,
+the size of the above files will not change.
+
+In new Openshift versions, there is a file that contains the file names,
```

Comment:
Maybe specify the version this was added just for reference? Not sure how much it matters, but "new" is relative and this document will be around for a while I assume.

Reply:
OK